### ISSUE TITLE
1st-party tracker blocking

### ISSUE THREAD

**Issue by gurgunday** [ghostery/ghostery-extension#476]
Since a week or so, cases of 1st-party tracking that seems to be unblockable started appearing. There have been discussions on places such as uBlock Origin's GitHub page. (More can be found [here](https://github.com/uBlockOrigin/uBlock-issues/issues/780))

Is Ghostery equipped to handle these first party tracking scripts and if not, is there anything that can be done to prevent this kind of tracking at all?

**Comment by remusao** [comment]
Hey @gurgunday!

We are following these developments very closely. We're still assessing what the best course of action and how big of a privacy issue this first-party DNS method is. But we will update you on the status ASAP. You can be assured that if something needs to be fixed we will push an update very soon.

Thanks for opening an issue!

Rémi

**Comment by gurgunday** [comment]
Seems like a [security concern](https://twitter.com/konarkmodi/status/1198541310807986176) as well as a privacy concern. Yet all the major tracker companies have started to support it.

**Comment by remusao** [comment]
> Seems like a [security concern](https://twitter.com/konarkmodi/status/1198541310807986176) as well as a privacy concern. Yet all the major tracker companies have started to support it.

Yes indeed. Not sure if you noticed but @konark-cliqz who wrote the tweet is a co-worker of mine at Ghostery/Cliqz and we are currently running an in-depth investigation of all negative consequences that 1st party tracking can have.

In parallel, we have already developed some tooling to automatically and continuously uncover usage of this pattern across the Internet to generate blocking rules that can be used by Ghostery and Cliqz to block this tracking; it is only a first-step.

We are also exploring ways to leverage our [search engine](https://beta.cliqz.com) to help discover 1st party tracking at scale. We already crawl the Web and have access to content of millions of domains in our index; this puts us in a unique position to run analysis *truly* at scale and in real-time to inform our tracking protection.

We will keep you posted about development soon but we expect protection against 1st tracking to ramp up in the coming days.

Cheers,
Rémi

**Comment by gurgunday** [comment]
Wow, I sure hadn't noticed that! Seems like Cliqz really has a great team that knows what they are doing. Thanks for your answer and as always, keep up the great work.

**Comment by gurgunday** [comment]
The latest [release](https://github.com/gorhill/uBlock/releases/tag/1.25.0) of uBlock Origin seems to have fixed this issue.

**Comment by remusao** [comment]
> The latest [release](https://github.com/gorhill/uBlock/releases/tag/1.25.0) of uBlock Origin seems to have fixed this issue.

Hey, sorry for not updating the ticket earlier. We have released this "fix" already in Cliqz browser last November and it is in the pipeline for next Ghostery release as well. In the meanwhile we also pushed custom rules to all Ghostery users. We analyze websites on the backend to detect cases of 1st party tracking and create custom rules to block these. The rules have been pushed last November as well so Ghostery users are protected, but the CNAME trick will be deployed too, soon. This CNAME trick only works on Firefox, though, whereas the custom rules that we already push also protect Ghostery users on Chrome.
